### PR TITLE
[Car Racing] Submit Muhammad Wahaj Khan's mission

### PR DESCRIPTION
**Pull request by WahajK**
### Summary:
This is my solution for the Car Racing Game mission. The program implements the car racing game according to the provided functional, input/output, and programming requirements.

### Key Features Implemented:
- **Car Movement**: Cars move forward based on randomly generated numbers.
- **Race Progression**: The race progresses round by round, displaying each car's movement.
- **Winners Determination**: After all rounds, the winners (cars with the maximum distance) are displayed.
- **Input Validation**: Input is validated for car names (ensuring they don’t exceed 5 characters) and rounds (ensuring positive integers are entered).
- **Test Cases**: A comprehensive set of tests has been created to verify the functionality of the application, including edge cases for input handling and car movement.

### Tests:
- **Game Logic**: Tests the correct movement of cars during a single round and multiple rounds.
- **Input Parser**: Tests the input parsing for car names and total rounds, including handling invalid input.
- **Output**: Ensures that the final winners are displayed correctly, based on the race results.
- **Exceptions**: Ensures that the program throws `IllegalArgumentException` for invalid inputs such as blank car names or non-numeric round counts.

### Assumptions & Decisions:
- **At least two cars**: I considered adding an assumption that a minimum of two cars must be entered for the race. However, since this was not explicitly stated in the requirements, I chose not to implement this assumption in the final code as I wasn’t sure whether it was allowed. I left it out to follow the stated requirements strictly.

Looking forward to the review. Thank you for your time and feedback!

### CODE REVIEW

File: src/main/kotlin/racingcar/GameLogic.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package racingcar
+
+import camp.nextstep.edu.missionutils.Randoms
+
+object GameLogic {
```
Comment (Jiihyun) [inline src/main/kotlin/racingcar/GameLogic.kt:5]: Why did you choose to declare objects instead of classes throughout the file?
I’m curious if there’s a specific reason or advantage you had in mind when deciding to use objects!

File: src/main/kotlin/racingcar/Car.kt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package racingcar
+
+data class Car(val name: String, var position: Int = 0)
```
Comment (Jiihyun) [inline src/main/kotlin/racingcar/Car.kt:3]: What made you decide to declare the `Car` class as a `data class`?

I understand that using a data class helps eliminate boilerplate code for methods like equals() and toString(), but it doesn’t seem like these features are being utilized in your code. So, I’m curious about any other advantages you've got by declaring `Car` class as a data class😄

File: src/main/kotlin/racingcar/GameLogic.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        return Randoms.pickNumberInRange(0, 9) >= MOVE_THRESHOLD
+    }
+
+    private fun moveCar(car: Car) {
+        if (checkMove()) {
+            car.position++
+        }
+    }
+
+    fun startRace(cars: List<Car>, rounds: Int) {
```
Comment (Jiihyun) [inline src/main/kotlin/racingcar/GameLogic.kt:18]: It would be better to organize functions with public first, and private later.
It could make the code more intuitive to navigate for readers :)

File: src/main/kotlin/racingcar/InputParser.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    fun readCarNames(): List<String> {
+        println("Enter car names (comma-separated):")
+        return parseCarNames(Console.readLine())
+    }
+
+    fun readTotalRounds(): Int {
+        println("Enter number of rounds:")
+        return parseTotalRounds(Console.readLine())
+    }
+}
```
Comment (Jiihyun) [inline src/main/kotlin/racingcar/InputParser.kt:35]: Don’t forget the EOF (end of file) newline!
It’s a small detail, but it can help improve consistency. If you’re using IntelliJ, here’s a helpful post on enabling this setting in IntelliJ:
https://stackoverflow.com/questions/16761227/how-to-make-intellij-idea-insert-a-new-line-at-every-end-of-file

**Review by Jiihyun — COMMENTED** [review]
Hi! I came across your code randomly and noticed that you welcome reviews, so I left a few comments😊

Best of luck with your next mission, and keep up the great work! 💪

**Comment by WahajK** [comment]
> Hi! I came across your code randomly and noticed that you welcome reviews, so I left a few comments😊
> 
> Best of luck with your next mission, and keep up the great work! 💪

Hey, apologies that I couldn't get back to you in time, thank you for your response. I couldn't follow up with it due to being busy with life, but I have read it now and will take the suggestions as good advice and try to incorporate them in my future works.
